src/groups.py

```diff
@@ -2,27 +2,27 @@
 # all dependancy on nodeattr is to be located in this file
 import config
 
+import click
 from plumbum import local, ProcessExecutionError
 from tempfile import NamedTemporaryFile
-from click import ClickException
 from re import search
 
 def list_groups():
     groups = __nodeattr(arguments=['-l'])
-    groups.remove('')
+    while '' in groups: groups.remove('')
```

Comment:
Why is this required? When are empty groups returned from `nodeattr`?

Reply:
As per IRL: this is an existing issue that's been dealt with here https://github.com/alces-software/adminware/pull/169/commits/bbac2171ffacf3ea31f301f914b842c2945a6ef9